integreat_cms/cms/forms/translations/machine_translation_form.py

```diff
@@ -0,0 +1,112 @@
+        widget=forms.CheckboxSelectMultiple(),
+        queryset=None,
+        required=False,
+        label=_("Update existing translations:"),
+    )
+
+    translations_to_create = forms.ModelMultipleChoiceField(
+        widget=forms.CheckboxSelectMultiple(),
+        queryset=None,
+        required=False,
+        label=_("Create new translations:"),
+    )
+
+    class Meta:
+        fields = [
+            "automatic_translation",
+            "translations_to_update",
+            "translations_to_create",
+        ]
+
+    def __init__(self, instance, region, language, **kwargs):
+        r"""
+        Initialize MT translation form
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+        """
+        super().__init__(**kwargs)
+
+        self.initial["automatic_translation"] = False
+
+        translation_targets = [
+            language
+            for language in region.language_tree_nodes.filter(parent=language.id)
+            if language.mt_provider and mt_to_lang_is_permitted(region, language.slug)
+        ]
+
+        # completely hide options if no target languages exist
+        if not translation_targets:
+            del self.fields["automatic_translation"]
+            return
+
+        to_update, to_create = [], []
+        for target in translation_targets:
+            target_type = (
+                to_update
+                if instance and instance.get_translation(target.slug)
+                else to_create
+            )
+            target_type.append((target.id, target.translated_name))
+
+        self.fields["translations_to_update"].choices = to_update
+        self.fields["translations_to_create"].choices = to_create
+
+        self.fields[
+            "translations_to_update"
+        ].queryset = region.language_tree_nodes.filter(
+            id__in=[language[0] for language in to_update]
+        )
+        self.fields[
+            "translations_to_create"
+        ].queryset = region.language_tree_nodes.filter(
+            id__in=[language[0] for language in to_create]
+        )
+
+        self.initial["translations_to_update"] = [language[0] for language in to_update]
+
+    def clean(self):
+        """
+        Validate form fields which depend on each other, see :meth:`django.forms.Form.clean`
+
+        :return: The cleaned form data
+        :rtype: dict
+        """
+        cleaned_data = super().clean()
+
+        cleaned_data["translation_target_languages"] = []
+        cleaned_data["translation_target_languages"].extend(
+            language.slug for language in cleaned_data.pop("translations_to_update", [])
+        )
+        cleaned_data["translation_target_languages"].extend(
+            language.slug for language in cleaned_data.pop("translations_to_create", [])
+        )
+
+        if not cleaned_data["automatic_translation"]:
+            cleaned_data["translation_target_languages"] = []
+        return cleaned_data
```

Comment:
Yeah, fair point. I guess it was working fine, but I see how it could lead to confusion. Plus I like your solution of calling `get_target_language_slugs` from the form views!